test/e2e-go/cli/goal/expect/statefulTealAppReadTest.exp

```diff
@@ -0,0 +1,145 @@
+    puts "calling goal app call with wrong scripts"
+    spawn goal app call --app-id $APP_ID --from $PRIMARY_ACCOUNT_ADDRESS -w $PRIMARY_WALLET_NAME --approval-prog-raw ${TEAL_PROGS_DIR}/wrongupgrade.tealc --clear-prog-raw ${TEAL_PROGS_DIR}/wrongupgrade.tealc -d $TEST_PRIMARY_NODE_DIR
+    expect {
+        timeout { puts timeout; ::AlgorandGoal::Abort  "\n Failed to see expected output" }
+        "Couldn't broadcast tx with algod: HTTP 400 Bad Request: TransactionPool.Remember: transaction*" {puts "received expected error"; close}
+        eof { close; ::AlgorandGoal::Abort "did not receive expected error" }
+    }
+
+    # Calling app as an update but with right scripts should succeed
+    spawn goal app update --app-id $APP_ID --from $PRIMARY_ACCOUNT_ADDRESS -w $PRIMARY_WALLET_NAME -d $TEST_PRIMARY_NODE_DIR --approval-prog-raw ${TEST_ROOT_DIR}/upgraded.tealc  --clear-prog-raw ${TEST_ROOT_DIR}/upgraded.tealc
+    expect {
+        timeout { puts timeout; ::AlgorandGoal::Abort  "\n Failed to see expected output" }
+        "Please enter the password for wallet '$PRIMARY_WALLET_NAME':" {send "$PRIMARY_WALLET_PASSWORD\r" ; exp_continue}
+        "*committed in round*" {puts "app update succeeded"; close}
+        eof {close; ::AlgorandGoal::Abort "app update failed" }
+    }
+
+    # Global state should be empty
+    spawn goal app read --guess-format --app-id $APP_ID --global -d $TEST_PRIMARY_NODE_DIR
+    expect {
+        timeout { puts timeout; ::AlgorandGoal::Abort  "\n Failed to see expected output" }
+        -re {^\{\}$} {puts "app read succeeded"}
+        eof {close; ::AlgorandGoal::Abort "app read failed" }
+    }
+
+    # Calling app should succeed
+    spawn goal app call --app-id $APP_ID --from $PRIMARY_ACCOUNT_ADDRESS -w $PRIMARY_WALLET_NAME -d $TEST_PRIMARY_NODE_DIR
+    expect {
+        timeout { puts timeout; ::AlgorandGoal::Abort  "\n Failed to see expected output" }
+        "Please enter the password for wallet '$PRIMARY_WALLET_NAME':" {send "$PRIMARY_WALLET_PASSWORD\r" ; exp_continue}
+        "*committed in round*" {puts "app call succeeded"}
+        eof {close; ::AlgorandGoal::Abort "app call failed" }
+    }
+
+    # Global state should now have 'foo': 'foo' key
+    spawn goal app read --guess-format --app-id $APP_ID --global -d $TEST_PRIMARY_NODE_DIR
+    expect {
+        timeout { puts timeout; ::AlgorandGoal::Abort  "\n Failed to see expected output" }
+        "*foo*" {puts "app read succeeded"}
+        eof {close; ::AlgorandGoal::Abort "app read failed" }
+    }
+
+    # Shutdown the network
+    ::AlgorandGoal::StopNetwork $NETWORK_NAME $TEST_ALGO_DIR $TEST_ROOT_DIR
+
+    puts "Goal statefulTealAppReadTest Successful"
+
+}
+
+
+if { [catch {
+    source  goalExpectCommon.exp
+
+    puts "starting statefulTealAppReadTest"
+
+    puts "TEST_ALGO_DIR: $TEST_ALGO_DIR"
+    puts "TEST_DATA_DIR: $TEST_DATA_DIR"
+
+    statefulTealAppReadTest  $TEST_ALGO_DIR  $TEST_DATA_DIR
```

Comment:
exceed spaces around TEST_ALGO_DIR?

Reply:
I will clean this up in a subsequent PR.